### PR TITLE
added function for closing popup modals

### LINKED ISSUE #57: [Login popup] The user closes Login popup by clicking outside of this popup
**Environment**: Windows10 21H2, Chrome 112.0.5615.138
**Reproducible**: always

**Preconditions**

The user enters Login popup window by clicking Login button  in the upper right corner of the header

**Steps to reproduce**

Click outside of the Login popup

**Actual result**

Login popup window closes

**Expected result**

Nothing should happen

### PR DESCRIPTION
**Pull request by ArtemHolikov**
- [x] Popup window closes after click outside of window (overlay).

https://github.com/Space2Study-Project/Space2Study-FrontEnd/assets/102384883/513c9750-4fac-449a-948b-89b80be94175

### CODE REVIEW

**Review by AndriiNasadiuk — APPROVED** [review]
Congratulations

**Comment by LeonidShamarin** [comment]
Yes, I see, nice

**Review by marianamartsy — APPROVED** [review]
great job!!! well done!

**Review by LeonidShamarin — APPROVED** [review]
good work

**Review by Mygal-Alex — APPROVED** [review]
good job

File: src/components/popup-dialog/PopupDialog.jsx [hunk 1]
```diff
@@ -27,7 +27,15 @@ const PopupDialog = ({
...
-  const handleClose = () => void onClose()
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  const handleCloseIcon = (event) => {
+    event.stopPropagation()
+    onClose()
+  }
```
Comment (vlad-khrychov) [inline src/components/popup-dialog/PopupDialog.jsx]: Can we use `handleCloseIcon` for both cases? O we need `stopPropagation` only for `IconButton`?

File: src/components/popup-dialog/PopupDialog.jsx [hunk 2]
```diff
@@ -36,7 +44,6 @@ const PopupDialog = ({
       disableRestoreFocus
       fullScreen={isMobile}
       maxWidth='xl'
-      onClick={(e) => e.stopPropagation()}
       onClose={handleClose}
```
Comment (vlad-khrychov) [inline src/components/popup-dialog/PopupDialog.jsx]: Can we delete handleClose and use `onClose` or `handleCloseIcon` here?

File: src/components/popup-dialog/PopupDialog.jsx [hunk 3]
```diff
@@ -27,7 +27,15 @@ const PopupDialog = ({
 
   const handleMouseOver = () => timerId && clearTimeout(timerId)
   const handleMouseLeave = () => timerId && closeModalAfterDelay()
-  const handleClose = () => void onClose()
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  const handleCloseIcon = (event) => {
```
Comment (vlad-khrychov) [inline src/components/popup-dialog/PopupDialog.jsx]:
```suggestion
  const handleClickIcon = (event) => {
```

**Review by vlad-khrychov — APPROVED** [review]

**Review by online-marathon — APPROVED** [review]

**Pull request merged**